llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -610,8 +610,11 @@ static Value *foldLogOpOfMaskedICmps(Value *LHS, Value *RHS, bool IsAnd,
       APInt NewMask = *ConstB & *ConstD;
       if (NewMask == *ConstB)
         return LHS;
-      if (NewMask == *ConstD)
+      if (NewMask == *ConstD) {
+        if (IsLogical)
+          cast<ICmpInst>(RHS)->setSameSign(false);
```

Comment:
as RHS is a Value it shall not be assumed that it is possible to cast it to ICmpInst

Reply:
yes so that I do not need to update it when I make that commit